opentelemetry-instrumentation/tests/test_semconv.py

```diff
@@ -0,0 +1,222 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from unittest import TestCase
+from unittest.mock import Mock, patch
+
+from opentelemetry.instrumentation._semconv import (
+    OTEL_SEMCONV_STABILITY_OPT_IN,
+    _OpenTelemetrySemanticConventionStability,
+    _OpenTelemetryStabilitySignalType,
+    _set_status,
+    _StabilityMode,
+)
+from opentelemetry.trace.status import StatusCode
+
+
+def semconv_stability_mode(mode):
+    def decorator(test_case):
+        @patch.dict(os.environ, {OTEL_SEMCONV_STABILITY_OPT_IN: mode})
+        def wrapper(*args, **kwargs):
+            _OpenTelemetrySemanticConventionStability._initialized = False
+            _OpenTelemetrySemanticConventionStability._initialize()
+            return test_case(*args, **kwargs)
+
+        return wrapper
+
+    return decorator
+
+
+class TestOpenTelemetrySemConvStability(TestCase):
+    @semconv_stability_mode("")
+    def test_default_mode(self):
+        self.assertEqual(
+            _OpenTelemetrySemanticConventionStability._get_opentelemetry_stability_opt_in_mode(
+                _OpenTelemetryStabilitySignalType.HTTP
+            ),
+            _StabilityMode.DEFAULT,
+        )
+
+    @semconv_stability_mode("http")
+    def test_http_mode(self):
+        self.assertEqual(
+            _OpenTelemetrySemanticConventionStability._get_opentelemetry_stability_opt_in_mode(
+                _OpenTelemetryStabilitySignalType.HTTP
+            ),
+            _StabilityMode.HTTP,
+        )
+
+    @semconv_stability_mode("http/dup")
+    def test_http_dup_mode(self):
+        self.assertEqual(
+            _OpenTelemetrySemanticConventionStability._get_opentelemetry_stability_opt_in_mode(
+                _OpenTelemetryStabilitySignalType.HTTP
+            ),
+            _StabilityMode.HTTP_DUP,
+        )
+
+    @semconv_stability_mode("database")
+    def test_database_mode(self):
+        self.assertEqual(
+            _OpenTelemetrySemanticConventionStability._get_opentelemetry_stability_opt_in_mode(
+                _OpenTelemetryStabilitySignalType.DATABASE
+            ),
+            _StabilityMode.DATABASE,
+        )
+
+    @semconv_stability_mode("database/dup")
+    def test_database_dup_mode(self):
+        self.assertEqual(
+            _OpenTelemetrySemanticConventionStability._get_opentelemetry_stability_opt_in_mode(
+                _OpenTelemetryStabilitySignalType.DATABASE
+            ),
+            _StabilityMode.DATABASE_DUP,
+        )
+
+    @semconv_stability_mode("database,http")
```

Comment:
Could we also have tests for mixed config, e.g. `database,http/dup` and `database_dup/http`

Reply:
Yes, sure. Will add

Reply:
Added. Thanks!